src/relative.ts

```diff
@@ -0,0 +1,49 @@
+import { format } from './format';
+import { normalizeParsed } from './normalize';
+import { parse } from './parse';
+
+export function relative(from: string, to: string): string {
+  const toParsed = normalizeParsed(parse(to));
+  // If `to` is already relative, just return that normalized
+  if (!toParsed.absolute) {
+    return format(toParsed);
+  }
+  const fromParsed = normalizeParsed(parse(from));
+  // If a relative URL is not possible, return an absolute one.
+  if (toParsed.origin !== fromParsed.origin) return format(toParsed);
+  if (!fromParsed.absolute) return format(toParsed);
+  if (fromParsed.drive !== toParsed.drive) return format(toParsed);
+
+  // Otherwise, remove the common parts.
+  const fromPath = fromParsed.path;
+  const toPath = toParsed.path;
+  // NOTE: 'from' should always be a directory. Otherwise, it doesn't make a lot of sense.
+  if (fromParsed.base) fromPath.push(fromParsed.base);
+  if (toParsed.base) toPath.push(toParsed.base);
+
+  // Toss away common path segments
+  const maxIter = Math.max(fromPath.length, toPath.length);
+  for (let i = 0; i < maxIter; i++) {
+    if (fromPath[0] === toPath[0]) {
+      fromPath.shift();
```

Comment:
This looks tricky. You might end up with out-of-bound access, at least according to the code.
Suggest you decrease `maxIter` or `i`.

Reply:
whaaat is   going on.
This should have been `const maxIter = Math.min(fromPath.length, toPath.length);` eh? Is that what I was going for you think?

Reply:
Having `Math.min` might not help here.
To illustrate run the following
```ts
const a = [0, 1]
const b = [0, 1];
const maxIter = Math.min(a.length, b.length);
for (let i = 0; i < maxIter; i++) {
    if (a[0] === b[0]) {
       a.shift();
       b.shift();
    } 

   console.assert(a[i] !== void 0, 'oops')
}
```
Obviously it doesn't happen under all circumstances.

Reply:
I never use `i`.

Reply:
So... it never happens?

Reply:
True, forgot about that. Math.min should do the trick then in such case.

Reply:
@wmhilton it may happen, just update the above example and change it to something as follows
```
const a = [0, 1]
const b = [0, 1, 2];
const maxIter = Math.max(a.length, b.length);
for (let i = 0; i < maxIter; i++) {
   console.assert(a[0] !== void 0, 'oops')

    if (a[0] === b[0]) {
       a.shift();
       b.shift();
    } 

}
```

Reply:
^ this assumes you keep Math.max

Reply:
In any case, the code just works. Would just add a check against out-of-bounds access as this tended to cause a de-opt in V8.

Reply:
right it is supposed to use `Math.min`